src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.TimeDate/Components/ResultHelper.cs

```diff
@@ -0,0 +1,222 @@
+                        Label = Resources.Microsoft_plugin_timedate_Year,
+                        IconType = ResultIconType.Date,
+                    },
+                    TimeAndDateHelper.GetCalendarEraResult(dateTimeNow, calendar),
+                    new AvailableResult()
+                    {
+                        Value = dateTimeNow.ToString("Y", CultureInfo.CurrentCulture),
+                        Label = Resources.Microsoft_plugin_timedate_MonthYear,
+                        IconType = ResultIconType.Date,
+                    },
+                    new AvailableResult()
+                    {
+                        Value = dateTimeNow.Ticks.ToString(CultureInfo.CurrentCulture),
+                        Label = timestamp == null ? Resources.Microsoft_plugin_timedate_WindowsFileTimeNow : Resources.Microsoft_plugin_timedate_WindowsFileTime,
+                        IconType = ResultIconType.DateTime,
+                    },
+                    new AvailableResult()
+                    {
+                        Value = dateTimeNowUtc.ToString("u"),
+                        Label = Resources.Microsoft_plugin_timedate_UniversalTime,
+                        IconType = ResultIconType.DateTime,
+                    },
+                    new AvailableResult()
+                    {
+                        Value = dateTimeNow.ToString("s"),
+                        Label = Resources.Microsoft_plugin_timedate_Iso8601,
+                        IconType = ResultIconType.DateTime,
+                    },
+                    new AvailableResult()
+                    {
+                        Value = dateTimeNowUtc.ToString("yyyy-MM-ddTHH:mm:ss", CultureInfo.InvariantCulture),
+                        Label = Resources.Microsoft_plugin_timedate_Iso8601Utc,
+                        IconType = ResultIconType.DateTime,
+                    },
+                    new AvailableResult()
+                    {
+                        Value = dateTimeNow.ToString("yyyy-MM-ddTHH:mm:ssK", CultureInfo.InvariantCulture),
+                        Label = Resources.Microsoft_plugin_timedate_Iso8601Zone,
+                        IconType = ResultIconType.DateTime,
+                    },
+                    new AvailableResult()
+                    {
+                        Value = dateTimeNowUtc.ToString("yyyy-MM-ddTHH:mm:ss'Z'", CultureInfo.InvariantCulture),
+                        Label = Resources.Microsoft_plugin_timedate_Iso8601ZoneUtc,
+                        IconType = ResultIconType.DateTime,
+                    },
+                    new AvailableResult()
+                    {
+                        Value = dateTimeNow.ToString("R"),
+                        Label = Resources.Microsoft_plugin_timedate_Rfc1123,
+                        IconType = ResultIconType.DateTime,
+                    },
+                });
+            }
+
+            return results;
```

Comment:
```cs
// return only results where value is not empty
// this can happen, for example, when we can't read the 'era' or when 'era == era abbreviation' and we set value explicit to an empty string.
return results.select(x => string.NotNullOrEmpty(x.value)).ToList();
```